scripts/asm-installer/install_asm

```diff
@@ -728,6 +733,47 @@ validate_istio_version() {
   fi
 }
 
+validate_asm_version() {
+  info "Checking existing ASM version(s)..."
+  local OUTPUT
+  OUTPUT="$(retry 3 ./istio-${RELEASE}/bin/istioctl version -o json)"
+  local FOUND_INVALID_VERSION
+  FOUND_INVALID_VERSION="false"
+  for version in $(echo "${OUTPUT}" | jq -r '.meshVersion[].Info.version'); do
+    if ! [[ "$version" =~ "asm" ]]; then
+      fatal "Cannot upgrade from version $version. Only upgrade from ASM distribution is supported."
+    fi
+
+    # if asm version found, pattern: 1.6.11-asm.1-586f900508ad482ed32b830dd15f6c54b32b93ed
+    local VERSION_MAJOR VERSION_MINOR VERSION_POINT VERSION_REV
+    IFS="." read VERSION_MAJOR VERSION_MINOR VERSION_POINT VERSION_REV <<< "$version"
+    VERSION_POINT="$(echo "$VERSION_POINT" | sed 's/-.*//')"
+    VERSION_REV="$(echo "$VERSION_REV" | sed 's/-.*//')"
+    if [[ "$VERSION_MAJOR" -ne 1 || "$VERSION_MINOR" -lt 6 || "$VERSION_MINOR" -ge "$MINOR" || \
+      ( "$VERSION_MINOR" -eq "$MINOR" && \
+        ( "$VERSION_POINT" -ge "$POINT" || \
+          ("$VERSION_POINT" -eq "$POINT" && "$VERSION_REV" -ge "$REV"))) ]]; then
```

Comment:
This seems like a good candidate to pull into a different function.

```
version_less_than_installed() {
  VERSION=$1
  ....
  return 0/1
}
```

Reply:
SG